### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by eiri**
## PR summary

This PR adds port of travis to github actions to the repo as a pre-req of removing travis CI 

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  SERVER_AUTH_TYPE: basic
+  SERVER_URL: http://127.0.0.1:5984
+  SERVER_USERNAME: admin
+  SERVER_PASSWORD: password
+jobs:
+  test:
+    strategy:
+      matrix:
+        java: ['8'] 
+        # java: ['8', '11', '17']
```
Comment (ricellis) [inline .github/workflows/test.yml:13]: I assume you are planning to re-enable the full matrix when the review is complete?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        distribution: 'temurin'
+        java-version: ${{ matrix.java }}
+        cache: 'maven'
+        check-latest: false
+    - name: Setup CouchDB
+      shell: bash
+      run: ${GITHUB_WORKSPACE}/scripts/setup_couch.sh
+    - name: Set maven version
+      if: github.event_name == 'create' && startsWith(github.ref, 'refs/tags/')
+      run: mvn versions:set -DnewVersion=${{ github.ref_name }} -DgenerateBackupPoms=false
```
Comment (ricellis) [inline .github/workflows/test.yml:39]: Not needed, this is leftover from the core's original travis templte, we do our version management stuff in Jenkins.
```suggestion
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Run tests
+      run: mvn --batch-mode verify
```
Comment (ricellis) [inline .github/workflows/test.yml:41]: I'm not opposed to using `batch-mode` here, but I didn't see any failure that seemed to necessitate it, was there a problem?

**Review by ricellis — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (eiri) [reply .github/workflows/test.yml:13]: yes

File: .github/workflows/test.yml (hunk 2 above)
Comment (eiri) [reply .github/workflows/test.yml:39]: so, just to be clear, we also do not publish from travis to `ossrh` and `na-artifactory` and I can ignore migration of  `.travis.settings.xml` too?

Comment (ricellis) [reply .github/workflows/test.yml]: Correct, you can ignore. The file is used in Jenkins so a bit of a misnomer.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: test
+on: [push, pull_request]
```
[the remark this answers is not in the document]
Comment (eiri) [reply .github/workflows/test.yml:2]: yes, I have a note to follow up on adding it to all the repos after we switch from travis

**Review by mojito317 — APPROVED** [review]
+1

Could you please refer the [follow up solution/issue](https://github.com/IBM/cloudant-java-sdk/pull/312/files#r924454690) to this PR later?

**Pull request merged**